website/src/content/development/forms.mdx

```diff
@@ -0,0 +1,673 @@
+      />
+    </Spacings.Stack>
+  );
+
+  return props.children({
+    formElements,
+    isDirty: formik.dirty,
+    isSubmitting: formik.isSubmitting,
+    submitForm: formik.handleSubmit,
+    handleCancel: formik.handleReset,
+  });
+}
+```
+
+The Channels pages then can be refactored as following:
+
+```js title="channels-create.js"
+import { useCallback } from 'react';
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import { FormModalPage } from '@commercetools-frontend/application-components';
+import { docToFormValues, formValuesToDoc } from './conversions';
+import ChannelsForm from './channels-form';
+
+const ChannelsCreate = (props) => {
+  const languages = useApplicationContext((context) => context.project.languages);
+  const handleSubmit = useCallback(
+    async (formValues) => {
+      const data = formValuesToDoc(formValues);
+      // This would trigger the request, for example a mutation.
+      const result = await createChannel(data);
+      // If successful, show a notification and redirect
+      // to the Channels details page.
+      // If errored, show an error notification.
+    },
+    [createChannel]
+  );
+
+  return (
+    <ChannelsForm
+      initialValues={docToFormValues(null, languages)}
+      onSubmit={handleSubmit}
+    >
+      {(formProps) => {
+        return (
+          <FormModalPage
+            title="Create a channel"
+            isOpen
+            onClose={props.onClose}
+            isPrimaryButtonDisabled={formProps.isSubmitting}
+            onSecondaryButtonClick={() => {
+              formProps.handleCancel();
+              props.onClose()
+            }}
+            onPrimaryButtonClick={formProps.submitForm}
+          >
+            {formProps.formElements}
+          </FormModalPage>
+        )
+      }}
+    </ChannelsForm>
+  );
+}
+```
+
+```js title="channels-details.js"
+import { useCallback } from 'react';
+import { useRouteMatch } from "react-router-dom";
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import { FormModalPage } from '@commercetools-frontend/application-components';
+import LoadingSpinner from '@commercetools-uikit/loading-spinner';
+import useChannels from './use-channels';
+import { docToFormValues, formValuesToDoc } from './conversions';
+import ChannelsForm from './channels-form';
+
+const ChannelsDetails = (props) => {
+  const match = useRouteMatch();
+  const languages = useApplicationContext((context) => context.project.languages);
+  const { data: channel } = useChannels(match.params.id)
+  const handleSubmit = useCallback(
+    async (formValues) => {
+      const data = formValuesToDoc(formValues);
+      // This would trigger the request, for example a mutation.
+      const result = await updateChannel(data);
+      // If successful, show a notification.
+      // If errored, show an error notification.
+    },
+    [updateChannel]
```

Comment:
Since this is a detailed example and all the dependencies are defined, I'm missing where this `updateChannel` function comes from.